From 53010f0eb1fbf491f7ff930a70d8133fc8b13630 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Thu, 23 May 2019 08:36:22 +0200
Subject: [PATCH] libscap: Add missing <sys/sysmacros.h> include for glibc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Newer glibc versions require including <sys/sysmacros.h> for makedev()
macro.  Without it, the build reports undefined function:

  [12/112] /usr/lib/ccache/bin/x86_64-pc-linux-gnu-gcc-8.3.0 -DGRPC_INCLUDE_IS_GRPCPP=1 -DHAS_CAPTURE -DK8S_DISABLE_THREAD -DPLATFORM_NAME=\"Linux\" -I/tmp/portage/dev-util/sysdig-0.26.0/work/sysdig-0.26.0/common  -DNDEBUG -march=x86-64 -mtune=k8 -mcx16 -msahf -msse3 --param l1-cache-size=64 --param l1-cache-line-size=64 --param l2-cache-size=512 -O2 -pipe -frecord-gcc-switches -Wall -MD -MT userspace/libscap/CMakeFiles/scap.dir/scap_fds.c.o -MF userspace/libscap/CMakeFiles/scap.dir/scap_fds.c.o.d -o userspace/libscap/CMakeFiles/scap.dir/scap_fds.c.o -c /tmp/portage/dev-util/sysdig-0.26.0/work/sysdig-0.26.0/userspace/libscap/scap_fds.c
  /tmp/portage/dev-util/sysdig-0.26.0/work/sysdig-0.26.0/userspace/libscap/scap_fds.c: In function ‘scap_get_device_by_mount_id’:
  /tmp/portage/dev-util/sysdig-0.26.0/work/sysdig-0.26.0/userspace/libscap/scap_fds.c:847:19: warning: implicit declaration of function ‘makedev’ [-Wimplicit-function-declaration]

and afterwards fails to link:

  scap_fds.c:(.text+0x2838): undefined reference to `makedev'
  collect2: error: ld returned 1 exit status

sysdig-CLA-1.0-signed-off-by: Michał Górny <mgorny@gentoo.org>
---
 userspace/libscap/scap_fds.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/userspace/libscap/scap_fds.c b/userspace/libscap/scap_fds.c
index 452353a68..6eb7cd160 100644
--- a/userspace/libscap/scap_fds.c
+++ b/userspace/libscap/scap_fds.c
@@ -49,6 +49,7 @@ limitations under the License.
 #include <errno.h>
 #include <netinet/tcp.h>
 #if defined(__linux__)
+#include <sys/sysmacros.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
 //#include <linux/sock_diag.h>
